Add support for Sverio Paperboard v1 board definition

Pajenicko opened the pull request:
Adds a board definition for the Sverio Paperboard v1 (ESP32-S3 based EPDIY-compatible driver board by Pajenicko).

- `src/board/epd_board_sverio_paperboard_v1.c` — board definition (`sverio_paperboard_v1`), TPS65185 + PCA9555 based, 16-bit LCD bus on the ESP32-S3
- Registered in `CMakeLists.txt` and `src/epd_board.h`

Tested with the dragon example on real hardware (ED097TC2, VCOM -1.56 V).

The board design files (schematic/gerbers) are not published yet — a link will be added here once they are released as open source.

Commit:
Add support for Sverio Paperboard v1 board definition

martinberlin commented:
So @Pajenicko what we do with this? You want to send me an additional board or we leave it as is and merge it?

martinberlin committed:
Merge branch 'main' into feature/add-sverio-paperboard-v1-support

Pajenicko commented:
Hello, damn, I forgot .. sure, will send you new board on tuesday, right after easter!

martinberlin commented:
If you can please pull again this branch and run a:
clang-format ./src/board/sverio_paperboard_v1.c

To fix automatically the C lang formatting. Then just push the changes so the CI checks will pass

Pajenicko committed:
Rename board file to epd_board_ prefix and apply clang-format

Pajenicko commented:
Hi, quick update: a new pre-flashed board has been shipped to you today, so you should receive it soon. Hopefully this one boots up correctly and you'll be able to verify the pinout against the PR.

Also pushed a follow-up commit addressing your feedback:
  - Renamed the file to epd_board_sverio_paperboard_v1.c (added the epd_board_ prefix)
  - Ran clang-format so CI should pass now

martinberlin commented:
Please open a new PR since main branch changed already since you added this. Then you can add this changes fresh over the last changes.
Then I will test again but your board is now working. Please add also a link to your open source board design files.

martinberlin commented:
Hey @Pajenicko can you please try merging main back in this branch and pushing changes?
Then we could run the tests and try it again. Since your updates didn’t changed any of the core files it should work.

martinberlin commented:
Dear @Pajenicko please take action on this no longer than 15 or as latest 20 June. I asked you before if you are willing to merge **main** back here or you prefer to re-make the PR on the latest updates.
Just take action so we finish with this one. I'm not a big fan of having PRs open over a lifetime for obvious reasons

Pajenicko committed:
Merge remote-tracking branch 'origin/main' into feature/add-sverio-paperboard-v1-support

Pajenicko committed:
Migrate sverio_paperboard_v1 board to new I2C driver API

Adapt to upstream changes from the merge of main:
- epd_board_init() takes an EpdInitConfig parameter (#452)
- legacy i2c driver replaced by epd_board_i2c helper (#475)
- pca9555/tps65185 calls now use i2c_master device handles

Pajenicko committed:
Make power good status logging informational

printPowerGoodStatus() is also called on successful power-up, so log at
INFO level instead of ERROR. Make it static to avoid leaking a global
symbol.

Pajenicko commented:
@martinberlin done:

  - Merged current `main` back into this branch (no conflicts).
  - Migrated the board file to the new APIs that landed in `main` since this PR was opened: the new `EpdInitConfig` parameter of `init` (#452)
  and the new i2c_master-based driver via the `epd_board_i2c` helper (#475) - without this the branch wouldn't compile against current 
  `main`.
  - Changed the power-good status logging to `ESP_LOGI` (it was logged with `ESP_LOGE` even on successful power-up, which made the log you 
  posted earlier look like a failure - `0xFA` actually means all rails are good).
  - Verified on real hardware after the merge: dragon example on the Paperboard v1 + ED097TC2, power rails come up fine and the refresh 
  timings match what you measured (`diff: 32ms, draw: 381ms, total: 428ms`). Builds checked for both esp32s3 and esp32 targets with IDF 
  v5.4.4.
  - The board design files are now published as open source (CC-BY-SA-4.0, derived from EPDIY v7): 
  https://github.com/Pajenicko/Paperboard-hardware/tree/main/paperboard-epdiy - KiCad project, schematic PDF, 3D models and fabrication 
  outputs.

martinberlin commented:
Small fix here just to pass clang validation, just install it on your machine so it's available in command line. Then just do:

clang-format -i /path/to/file.c

OR: 
2. Integrate with VS Code (Recommended)
To stop fighting formatting errors manually, configure VS Code to use the exact ESP-Clang binary you provided.
Install the official Clang-Format extension by Xaver Hellauer.
Open your project's .vscode/settings.json file.
Add the following configuration, pointing directly to your ESP-IDF clang-format executable:
JSON
{
    "editor.formatOnSave": true,
    "[{cpp,c}]": {
        "editor.defaultFormatter": "xaver.clang-format"
    },
    "clang-format.executable": "/opt/esp/tools/esp-clang/esp-18.1.2_20240912/esp-clang/bin/clang-format"
}
💡 Tip: Setting "editor.formatOnSave": true ensures that every time you hit save, your code is instantly aligned with the ESP-IDF style guidelines.

Pajenicko committed:
Fix clang-format check

martinberlin commented:
@vroland this is now ready. About the board maybe is possible to make an "external board" section of boards that are not epdiy but built in the same fashion and license?

> https://github.com/Pajenicko/Paperboard-hardware/tree/main/paperboard-epdiy - KiCad project, schematic PDF, 3D models and fabrication

vroland commented:
Thanks! I don't have a way to test it so I'll just approve. We can definitely add a section with the link in the readme.
